src/nvim/option.c

```diff
@@ -1607,16 +1607,17 @@ do_set (
                              || (s == newval
                                  && arg[2] != '\\')))
 #endif
-                    )
-                  ++arg;                        /* remove backslash */
-                if (has_mbyte
-                    && (i = (*mb_ptr2len)(arg)) > 1) {
-                  /* copy multibyte char */
+                    ) {
+                  arg++;   // remove backslash
+                }
+                if ((i = mb_ptr2len(arg)) > 1) {
+                  // copy multibyte char
                   memmove(s, arg, (size_t)i);
                   arg += i;
                   s += i;
-                } else
+                } else {
                   *s++ = *arg++;
```

Comment:
I think this branch can be removed. The optimizer can already reason `memmove`/`memcpy`, and it's not like this code is super hot anyway...